plugins/core/hu.bme.mit.gamma.ui/src/hu/bme/mit/gamma/ui/taskhandler/OptimizerAndVerificationHandler.java

```diff
@@ -248,9 +249,12 @@ public void execute(Verification verification) throws IOException, InterruptedEx
 			// Traces have not been serialized yet, doing it now
 			verificationHandler.optimizeTraces();
 		}
+    
 		if (serializeTraces) {
 			verificationHandler.serializeTraces(); // Serialization in one pass
+      MakefileGenerator.tests.clear(); // Manual reset
```

Comment:
This is not a good pattern. Make sure to include the relevant C-related constructs only in the corresponding methods. Maybe the `MakefileGenerator` should be an instantiable class and then, these clears would not need to be carried out.